src/axom/primal/operators/in_polygon.hpp

```diff
@@ -0,0 +1,153 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+/*!
+ * \file in_polygon.hpp
+ *
+ * \brief Consists of methods that test whether a given query point is
+ * inside a given polygon.
+ *
+ * Uses a ray casting algorithm
+ */
+
+#ifndef AXOM_PRIMAL_IN_POLYGON_HPP_
+#define AXOM_PRIMAL_IN_POLYGON_HPP_
+
+// Axom includes
+#include "axom/config.hpp"
+
+#include "axom/primal/geometry/Point.hpp"
+#include "axom/primal/geometry/Polygon.hpp"
+
+// C++ includes
+#include <cmath>
+
+namespace axom
+{
+namespace primal
+{
+/*!
+ * \brief Computes the winding number for a point and a polygon
+ *
+ * \param [in] R The query point to test
+ * \param [in] P The Polygon object to test for containment
+ * \param [in] strict If true, points on the boundary are considered exterior.
+ * \param [in] EPS The tolerance level for collinearity
+ * 
+ * Uses an adapted ray-casting approach that counts quarter-rotation
+ * of vertices around the query point. 
+ *
+ * Directly uses algorithm in 
+ * Kai Hormann, Alexander Agathos, "The point in polygon problem for arbitrary polygons"
+ * Computational Geometry, Volume 20, Issue 3, 2001,
+ * 
+ * \return The integer winding number
+ */
+template <typename T>
+int winding_number(const Point<T, 2>& R,
+                   const Polygon<T, 2>& P,
+                   const bool strict = false,
+                   const double EPS = 1e-8)
+{
+  const int nverts = P.numVertices();
+
+  // If the query is a vertex, return a value interpreted
+  //  as "inside" by evenodd or nonzero protocols
+  if(axom::utilities::isNearlyEqual(P[0][0], R[1], EPS) &&
```

Comment:
Minor suggestion (feel free to ignore):

If you add the following to the top of the function, you can remove the namespaces a bit.
e.g.
```cpp
// at the top
using axom::utilities::isNearlyEqual;
using axom::numerics::determinant;
```

Then later, you can write:
```cpp
   if(isNearlyEqual(P[j][1], R[1], EPS)) {...}
   ...
   det = determinant(P[i][0] - R[0], P[j][0] - R[0],
                                P[i][1] - R[1], P[j][1] - R[1]);
```

Reply:
I'll keep it as is for this PR, but I'll definitely keep this in mind for the future. Thank you!